### ISSUE TITLE
Proposal: Method to detect RPi platform

### ISSUE THREAD

**Issue by wimrijnders** [mn416/QPULib#32]
This answer to a related question looks like a good way to detect if you're running on a RPi: https://raspberrypi.stackexchange.com/a/61071/61753.

If tried this on my two active RPi's:

```
> cat /sys/firmware/devicetree/base/model
Raspberry Pi 3 Model B Rev 1.2

> cat /sys/firmware/devicetree/base/model 
Raspberry Pi Model B Rev 2
```
*(I have a Pi 2 as well, which I could start up again for testing)*

Doing this on an intel machine:
```
> cat /sys/firmware/devicetree/base/model
cat: /sys/firmware/devicetree/base/model: No such file or directory
```

What I like about this, is that it's file-based, thus easy to implement over all platforms. Would like to hear your thoughts.

**Comment by wimrijnders** [comment]
Next I have to think about how to get the compilation to fail using the above.

The easy way out is to detect the platform in the makefile and block compilation altogether there. Will also need to think it out, but it's easier.

**Comment by wimrijnders** [comment]
Resolved by #38
